test/server/config_validation/server_test.cc

```diff
@@ -0,0 +1,40 @@
+#include "server/config_validation/hot_restart.h"
+#include "server/config_validation/server.h"
+
+#include "test/integration/server.h"
+#include "test/mocks/server/mocks.h"
+#include "test/mocks/stats/mocks.h"
+#include "test/test_common/environment.h"
+
+namespace Envoy {
+namespace Server {
+
+// Test param is the path to the config file to validate.
+class ValidationServerTest : public testing::TestWithParam<std::string> {
+protected:
+  ValidationServerTest()
+      : options_(GetParam()),
+        instance_(options_, restarter_, store_, access_log_lock_, component_factory_, local_info_) {
+  }
+
+  ~ValidationServerTest() { instance_.shutdown(); }
+
+  testing::NiceMock<MockOptions> options_;
+  ValidationHotRestart restarter_;
+  Stats::IsolatedStoreImpl store_;
+  Thread::MutexBasicLockable access_log_lock_;
+  TestComponentFactory component_factory_;
+  testing::NiceMock<LocalInfo::MockLocalInfo> local_info_;
+  ValidationInstance instance_;
+};
+
+TEST_P(ValidationServerTest, Foobar) {}
+
+INSTANTIATE_TEST_CASE_P(ValidConfigs, ValidationServerTest,
```

Comment:
It might be a little more complicated, because we want to run config tests on real configs off box that we will then use in production. This means potentially mocking out some of the TLS context stuff since we can't read the actual cert. (Or maybe we have some type of dummy cert that we can load somehow). I would put in relevant TODOs.